src/windows7.rs

```diff
@@ -0,0 +1,37 @@
+//! Implementation for Windows 7 and 8
+//!
+//! For targets where we cannot use ProcessPrng (added in Windows 10), we use
+//! RtlGenRandom. See windows.rs for a more detailed discussion of the Windows
+//! RNG APIs (and why we don't use BCryptGenRandom). On versions prior to
+//! Windows 10, this implementation works, while on Windows 10 and later, this
+//! is a thin wrapper around ProcessPrng.
+//!
+//! This implementation will not work on UWP targets, but those targets require
+//! Windows 10 regardless, so can use the ProcessPrng implementation.
+use crate::Error;
+use core::{ffi::c_void, mem::MaybeUninit};
+
+// This code is based on that produced by windows-bindgen with the APIs:
+//   Windows.Win32.Foundation.TRUE
+//   Windows.Win32.Security.Authentication.Identity.RtlGenRandom
+// but we avoid using windows-targets as it doesn't support older Windows.
+#[link(name = "advapi32")]
+extern "system" {
+    #[link_name = "SystemFunction036"]
+    fn RtlGenRandom(randombuffer: *mut c_void, randombufferlength: u32) -> BOOLEAN;
+}
+#[repr(transparent)]
+#[derive(PartialEq, Eq)]
+pub struct BOOLEAN(pub u8);
+pub const TRUE: BOOLEAN = BOOLEAN(1u8);
```

Comment:
I think this looks... redundant? We can just use `u8` for `RtlGenRandom` like we did before. Same for the `ProcessPrng` signature.

Reply:
Unfortunate but fine, IMO.

Reply:
Moved to the easier to read:
```rust
type BOOLEAN = u8;
const TRUE: BOOLEAN = 1u8;
```
as this code wasn't directly generated by `windows-bindgen` anyway.